### ISSUE TITLE
<meta charset> is (mostly) redundant

### ISSUE THREAD

**Issue by ndorfin** [joshbuchea/HEAD#186]
Hi!

Most hosts serve `text/html` with the `UTF-8` charset. Having a `<meta charset="UTF-8">` element is therefore redundant.

May I suggest adding a comment saying that declaring a charset is optional, but it's useful if users save the HTML to their device or the author wants to just work off the file system and not a host?

-S

**Comment by scottaohara** [comment]
Hi @ndorfin,

In spite of its perceived redundancy, I'd submit that the cost benefit of continuing to promote the usage of the tag far out weighs the consequences of a situation where a `utf-8` declaration is needed, but is lacking.

For instance, the HTML 5 specification states that declaring a `doctype` or wrapping one's documents in `<html>` tags is optional.  Most of the time, browsers will take care of the lack of these elements and things are just peachy.  But in the situations where the lack of these elements cause a failure, the user is the one that suffers.  Same with `meta charset="utf-8"`.

TLDR; setting the tag as optional sends the message that it has no consequences if left out. This is only true until it isn't, in which case things get messy.

**Comment by ndorfin** [comment]
All good, thanks.

I worry about the cargo-cult programming-ness of this and other patterns people follow, without fully understanding why they should or shouldn't be used.

Maybe it's worthwhile mentioning _why_ it's useful to add then?

```
<!-- Set the character encoding for this document, so that all characters within the UTF-8 space (such as emoji) are rendered correctly -->
```

**Comment by scottaohara** [comment]
Heh, so we're coming from very similar mindsets then.  :)

I like what you got there, and if you want to open a PR to add that comment above the instance of `"utf-8"` in the [Elements](https://github.com/joshbuchea/HEAD#elements) portion of the document, that'd be great.  

If you wanted, I'd be in favor of relocating that first comment out of the code block, the `<!-- Meta tags provide information for how a document should...`, and instead having that be a paragraph right before the code block.  That wan your new comment can be the first line of the code block.

Sound good?

**Comment by ndorfin** [comment]
Cool, OK :)
